collection.go

```diff
@@ -198,6 +198,17 @@ func NewCollectionWithOptions(spec *CollectionSpec, opts CollectionOptions) (col
 	}
 
 	for progName, origProgSpec := range spec.Programs {
+		shouldLoad := true
+		for _, exProg := range excludedPrograms {
+			if progName == exProg {
+				shouldLoad = false
+				break
+			}
+		}
```

Comment:
Nit, this might be cleaner if we extract a  `func stringInSlice(string, []string)`